encrypt.go

```diff
@@ -7,18 +7,18 @@ import (
 	"bytes"
 	"crypto/sha512"
 	"encoding/hex"
+	"fmt"
 	"io"
 
 	"golang.org/x/crypto/nacl/secretbox"
 )
 
 type encryptStream struct {
+	version    Version
```

Comment:
went ahead and replaced `header` with `version` (and fixed the single test that relied on it)